internal/venice-common/src/main/java/com/linkedin/venice/pubsub/PubSubPositionFactory.java

```diff
@@ -37,24 +37,16 @@ public PubSubPositionFactory(int positionTypeId) {
    * <p>
    * Internally, this delegates to {@link #fromPositionRawBytes(ByteBuffer)} to perform the actual decoding.
    *
-   * @param positionWireFormat the wire format containing the type ID and raw encoded bytes
+   * @param wireFormat the wire format containing the type ID and raw encoded bytes
    * @return a new {@link PubSubPosition} instance reconstructed from the wire format
    * @throws VeniceException if the type ID does not match the factory's expected type
    */
-  public PubSubPosition fromPositionRawBytes(PubSubPositionWireFormat positionWireFormat) {
-    if (positionWireFormat.getType() != positionTypeId) {
+  public PubSubPosition fromWireFormat(PubSubPositionWireFormat wireFormat) {
```

Comment:
api name/signature changes break the dependent projects. one way could be to introduce the new api without removing the old, and annotating as `@ApiStatus.ScheduledForRemoval`

Reply:
Updated.